githubutil/githubutil.go

```diff
@@ -114,22 +114,41 @@ func FetchEachPage(f func(options github.ListOptions) (*github.Response, error))
 
 // UploadFile is a function that will upload a file to a given repository.
 func UploadFile(ctx context.Context, client *github.Client, owner, repo, branch, path, message string, content []byte) error {
-	_, _, err := client.Repositories.CreateFile(ctx, owner, repo, path, &github.RepositoryContentFileOptions{
-		Message: &message,
-		Content: content,
-		Branch:  &branch,
+	err := Retry(func() error {
+		_, _, err := client.Repositories.CreateFile(ctx, owner, repo, path, &github.RepositoryContentFileOptions{
+			Message: &message,
+			Content: content,
+			Branch:  &branch,
+		})
+
+		return err
 	})
+
 	return err
 }
 
-// FileExists is a function that will check if a file exists in a given repository.
-func FileExists(ctx context.Context, client *github.Client, owner, repo, filePath string) (bool, error) {
-	_, _, resp, err := client.Repositories.GetContents(ctx, owner, repo, filePath, nil)
-	if err != nil {
+// DownloadFile is a function that will download a file from a given repository.
+func DownloadFile(ctx context.Context, client *github.Client, owner, repo, branch, path string) (file []byte, exists bool, err error) {
```

Comment:
I actually think this should return `file []byte, err error`, and return `nil, githubutil.ErrNotExists` if it doesn't exist. Then, the name `DownloadFile` matches the behavior better: callers that actually want to download the file only need to check `err != nil`. Only callers that care *why* the file couldn't be downloaded then need to check `errors.Is(err, githubutil.ErrNotExists)`. This also fits in with APIs like `os.ReadFile`.

Reply:
You're right, I'll make sure that I'm implementing this.